src/sentry/static/sentry/app/stores/configStore.tsx

```diff
@@ -33,33 +36,39 @@ const configStoreConfig: Reflux.StoreDefinition & ConfigStoreInterface = {
       ...this.config,
       [key]: value,
     };
-    this.trigger({[key]: value});
+    if (key === 'user') {
+      this.config.theme = this.getTheme();
+    }
+    this.trigger();
   },
 
   /**
    * This is only called by media query listener so that we can control
    * the auto switching of color schemes without affecting manual toggle
    */
   updateTheme(theme) {
-    if (this.config.user?.options.theme !== 'system') {
-      return;
+    this.systemTheme = theme;
+    if (this.config.user?.options.theme === 'system' && theme !== this.config.theme) {
+      this.set('theme', theme);
     }
+  },
 
-    this.set('theme', theme);
+  getTheme() {
+    // options?. to work around tests
+    const themeOption = this.config.user?.options?.theme;
+    return (themeOption === 'system' ? this.systemTheme : themeOption) || 'light';
   },
 
   getConfig() {
     return this.config;
   },
 
   loadInitialData(config): void {
-    const shouldUseDarkMode = config.user?.options.theme === 'dark';
-
     this.config = {
       ...config,
       features: new Set(config.features || []),
-      theme: shouldUseDarkMode ? 'dark' : 'light',
     };
+    this.config.theme = this.getTheme();
```

Comment:
Can we define this where we initialize `this.config`?

Reply:
the problem is that getTheme is unpure, it needs user to be set. upon consideration i think it could actually be a pure function that takes user though.

Reply:
i've made getTheme a pure function and moved it out of the class to address this issue. thank you for the feedback.